In [1]:
import scipy.stats as sts
from scipy.stats import norm
import math
import numpy as np
from scipy import stats

**Problem Statement 1:**

Blood glucose levels for obese patients have a mean of 100 with a standard deviation of 15.A researcher thinks that a diet high in raw cornstarch will have a positive effect on blood glucose levels. A sample of 36 patients who have tried the raw cornstarch diet have a mean glucose level of 108. Test the hypothesis that the raw cornstarch had an effect or not.

In [2]:
p_mean = 100
p_std = 15
n = 36
sample_mean = 108
alpha = 0.05

SE = p_std/n**0.5
print(f"SE: {SE}")
Z = (sample_mean-p_mean)/SE
print(f"Z_score: {Z}")
# by looking at z- table and p-value associated with 3.20 is 0.9993 
# The probability of having value less than 108 is 0.9993 and more than or equals to 108 is (1-0.9993)=0.0007.

#Step-4: Sice the probability of having mean glucose level more than or equals to 108  is 0.0007 which is  less than 0.05 
#so we will reject the Null hypothesis i.e. there is raw cornstarch effect.

SE: 2.5
Z_score: 3.2


**Problem Statement 2:**

In one state, 52% of the voters are Republicans, and 48% are Democrats. In a second state, 47% of the voters are Republicans, and 53% are Democrats. Suppose a simple random sample of 100 voters are surveyed from each state. What is the probability that the survey will show a greater percentage of Republican voters in the second state than in the first state?

H0 : p_1 >= p_2

H1 : p_1<p_2

In [3]:
def get_zscore(p_1,p_2,n1,n2,D=0): #D = difference in population proportion
    numerator = (p_1-p_2)-D
    denominator = np.sqrt(p_1*(1-p_1)/n1 +p_2*(1-p_2)/n2)
    z_value = numerator/denominator
    return z_value

In [4]:
p_1 = 0.52
p_2 = 0.47
n1 = 100
n2=100
z_score = get_zscore(p_1,p_2,n1,n2)

In [5]:
z_score

0.7080278164104213

In [6]:
p_value=1 - stats.norm.cdf(abs(z_score))
print("Probability of greater percentage of Republican voters in second state than in first state: {}".format(p_value))

Probability of greater percentage of Republican voters in second state than in first state: 0.23946399182220013


**Problem Statement 3:**

You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard deviation is 209. How well did you score on the test compared to the average test taker?

In [7]:
def get_zscore_1(mu,sigma,x_bar,n):
    return (x_bar-mu)/(sigma/np.sqrt(n))

In [8]:
z_score = get_zscore_1(1026,209,1100,n=1)
z_score

0.35406698564593303

I scored 0.35 standard deviation time above than the average test taker